go-controller/pkg/ovn/ovn.go

```diff
@@ -521,10 +534,28 @@ func (oc *Controller) WatchNodes() error {
 				logrus.Errorf(err.Error())
 				gatewaysFailed.Store(node.Name, true)
 			}
+
 		},
 		UpdateFunc: func(old, new interface{}) {
 			oldNode := old.(*kapi.Node)
 			node := new.(*kapi.Node)
+
+			oldNoHostSubnet, err := oc.noHostSubnet(oldNode)
```

Comment:
@JacobTanenbaum let's pull this out into a separate function that returns an error, and if the error != nil log the error and reutrn. Otherwise continue. This would be clearer than putting it all into UpdateFunc I think.

Reply:
fixed